xla/service/gpu/nccl_collective_broadcast_thunk.h

```diff
@@ -0,0 +1,75 @@
+/* Copyright 2024 The OpenXLA Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef XLA_SERVICE_GPU_NCCL_COLLECTIVE_BROADCAST_THUNK_H_
+#define XLA_SERVICE_GPU_NCCL_COLLECTIVE_BROADCAST_THUNK_H_
+
+#include <cstdint>
+
+#include "xla/service/collective_ops_utils.h"
+#include "xla/service/gpu/nccl_collective_thunk.h"
+#include "xla/service/gpu/nccl_api.h"
+
+namespace xla {
+namespace gpu {
+// Thunk that performs a NCCL-based collective broadcast.
+class NcclCollectiveBroadcastStartThunk : public NcclCollectiveThunk {
+ public:
+
+  static Status CheckImplementable(mlir::lmhlo_gpu::CollectiveBroadcastStartOp op,
```

Comment:
constructors from MLIR can be removed, I plan to start deleting LMHLO soon

Reply:
I've removed all methods that use `lmhlo_gpu::CollectiveBroadcastStartOp`.